In [4]:
import numpy as np
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/raditya2010631170111/df_20111/main/Parkinsson%20disease.csv")
df.tail()

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)  MDVP:Jitter(%)  \
190  phon_R01_S50_2      174.188       230.978        94.261         0.00459   
191  phon_R01_S50_3      209.516       253.017        89.488         0.00564   
192  phon_R01_S50_4      174.688       240.005        74.287         0.01360   
193  phon_R01_S50_5      198.764       396.961        74.904         0.00740   
194  phon_R01_S50_6      214.289       260.277        77.973         0.00567   

     MDVP:Jitter(Abs)  MDVP:RAP  MDVP:PPQ  Jitter:DDP  MDVP:Shimmer  ...  \
190           0.00003   0.00263   0.00259     0.00790       0.04087  ...   
191           0.00003   0.00331   0.00292     0.00994       0.02751  ...   
192           0.00008   0.00624   0.00564     0.01873       0.02308  ...   
193           0.00004   0.00370   0.00390     0.01109       0.02296  ...   
194           0.00003   0.00295   0.00317     0.00885       0.01884  ...   

     Shimmer:DDA      NHR     HNR  status      RPDE       DFA   spread1  \
190      0.07008  0.02764  19.517       0  0.448439  0.657899 -6.538586   
191      0.04812  0.01810  19.147       0  0.431674  0.683244 -6.195325   
192      0.03804  0.10715  17.883       0  0.407567  0.655683 -6.787197   
193      0.03794  0.07223  19.020       0  0.451221  0.643956 -6.744577   
194      0.03078  0.04398  21.209       0  0.462803  0.664357 -5.724056   

      spread2        D2       PPE  
190  0.121952  2.657476  0.133050  
191  0.129303  2.784312  0.168895  
192  0.158453  2.679772  0.131728  
193  0.207454  2.138608  0.123306  
194  0.190667  2.555477  0.148569  

[5 rows x 24 columns]

Missing Value

In [ ]:
df.isnull().any()

name                False
MDVP:Fo(Hz)         False
MDVP:Fhi(Hz)        False
MDVP:Flo(Hz)        False
MDVP:Jitter(%)      False
MDVP:Jitter(Abs)    False
MDVP:RAP            False
MDVP:PPQ            False
Jitter:DDP          False
MDVP:Shimmer        False
MDVP:Shimmer(dB)    False
Shimmer:APQ3        False
Shimmer:APQ5        False
MDVP:APQ            False
Shimmer:DDA         False
NHR                 False
HNR                 False
status              False
RPDE                False
DFA                 False
spread1             False
spread2             False
D2                  False
PPE                 False
dtype: bool

Menghapus fitur ID, Axis=Column, X=predictor=dataku1, Y=target=t

In [ ]:
X = df.drop(['name', 'status'],axis=1)
Y = df[['status']]
Y.tail()

status
190       0
191       0
192       0
193       0
194       0

Tdk ada fitur numerik untuk diencode

In [ ]:
X.dtypes

MDVP:Fo(Hz)         float64
MDVP:Fhi(Hz)        float64
MDVP:Flo(Hz)        float64
MDVP:Jitter(%)      float64
MDVP:Jitter(Abs)    float64
MDVP:RAP            float64
MDVP:PPQ            float64
Jitter:DDP          float64
MDVP:Shimmer        float64
MDVP:Shimmer(dB)    float64
Shimmer:APQ3        float64
Shimmer:APQ5        float64
MDVP:APQ            float64
Shimmer:DDA         float64
NHR                 float64
HNR                 float64
RPDE                float64
DFA                 float64
spread1             float64
spread2             float64
D2                  float64
PPE                 float64
dtype: object

seleksi fitur krn prediktor terlalu = lama proses

MDVP:Fo = threshold/batas=0 agar bisa tau Fo
10% konstan = dihapus Row = harus bervariasi

-  mencari quasi-konstan = nilai variance threshold > 0
misal threshold=0.1/10% = mencari dari 90% data
apabila data ada yg mirip2 akan tdk dihapus
apabila lbh dari 10% dihapus

- if 0/100% = hapus semuanya

174.188 = bisa konstan

True = tdk konstan/high variance
False = konstan/low variance

In [ ]:
from sklearn.feature_selection import VarianceThreshold
var_threshold = VarianceThreshold(threshold = 0) #Menghapus fitur konstan
var_threshold.fit(X)
var_threshold.get_support()

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True])

In [ ]:
var_threshold = VarianceThreshold(threshold = 0.1) #Menghapus fitur quasi-konstan
var_threshold.fit(X)
var_threshold.get_support()

array([ True,  True,  True, False, False, False, False, False, False,
       False, False, False, False, False, False,  True, False, False,
        True, False,  True, False])

versi panjang

In [ ]:
low_var_cols_another = []

for column in X.columns:
    if column X.columns[var_threshold.get_support()]:
      low_var_cols_another.append(column)

for col in low_var_cols_another:
    print(col, end=', ')

versi pendek

In [ ]:
low_var_cols = [column for column in X.columns
         if column not in X.columns[var_threshold.get_support()]]
for col in low_var_cols:
    print(col)

MDVP:Jitter(%)
MDVP:Jitter(Abs)
MDVP:RAP
MDVP:PPQ
Jitter:DDP
MDVP:Shimmer
MDVP:Shimmer(dB)
Shimmer:APQ3
Shimmer:APQ5
MDVP:APQ
Shimmer:DDA
NHR
RPDE
DFA
spread2
PPE


In [ ]:
X_high = X.drop(low_var_cols, axis=1)

In [ ]:
X_high.sample(5)

MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)     HNR   spread1        D2
87       176.281       227.381       125.610  13.922 -5.515071  3.136550
172      110.739       113.597       100.139  26.550 -6.085567  1.889002
32       198.383       215.203       193.104  30.775 -7.067931  1.512275
15       142.167       217.455        83.159  25.175 -5.340115  2.205546
167      260.105       264.919       237.303  21.083 -7.517934  1.881767

In [ ]:
var_threshold.fit(X_high)
var_threshold.get_support()

array([ True,  True,  True,  True,  True,  True])

In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.feature_selection import chi2

df=pd.read_csv('BankChurners.csv')
df.head()

In [ ]:
df.shape

In [ ]:
X = 
df.drop(["Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_1", "Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_2"], axis=1)
X.head()

In [ ]:
X.isnull().any()

In [ ]:
dataku2 = X.drop('CLIENTNUM', axis=1)
dataku2.head()

In [ ]:
dataku2.dtypes